src/fetch-url.js

```diff
@@ -1,14 +1,22 @@
 import { config as configStore } from './stores';
 import { isMapboxUrl, normalizeMapboxUrl } from './mapbox-urls';
 
-let mapboxGlAccessToken;
-configStore.subscribe(value => ({ mapboxGlAccessToken } = value));
+let mapboxGlAccessToken, maptilerApiKey;
+configStore.subscribe(
+  value => ({ mapboxGlAccessToken, maptilerApiKey } = value)
+);
 
 const fetchUrl = async url => {
   const urlIsMapbox = isMapboxUrl(url);
-  let nextUrl = url;
+  let nextUrl = new URL(url);
   if (urlIsMapbox) {
-    nextUrl = normalizeMapboxUrl(url, mapboxGlAccessToken);
+    nextUrl = normalizeMapboxUrl(nextUrl.toString(), mapboxGlAccessToken);
+  } else if (
+    maptilerApiKey &&
+    nextUrl.host === 'api.maptiler.com' &&
+    !nextUrl.searchParams.has('key')
+  ) {
+    nextUrl.searchParams.append('key', maptilerApiKey);
```

Comment:
I'm seeing Maptiler URLs fail in other renderers (Mapbox and Maplibre).

I think it's because here you're following the same pattern we use for a mapbox URL and as I think on this, I wonder if we need a change in this PR for Maptiler and a followup for Mapbox URLs.

Basically we haven't run into this with Mapbox URLs (with the mapbox protocol `mapbox://`) because we've considered them only valid via the Mapbox renderer despite us normalizing the URL here. This is because they typically also reference additional Mapbox protocol URLs for sprites and glyphs so we didn't consider this a bug (You'll see Mapbox styles using the protocol fail in Maplibre for example).

Maptiler (correct me if I'm wrong) is different in that it does not use a custom protocol inside the style and instead creates a valid url with the key present when requesting the style url with that key: 
```
"glyphs": "[https://api.maptiler.com/fonts/{fontstack}/{range}.pbf?key=fBwWvQI476rYQ6q9LrJf](https://api.maptiler.com/fonts/%7Bfontstack%7D/%7Brange%7D.pbf?key=fBwWvQI476rYQ6q9LrJf)"
```

So basically, we should be able to request the style with the new `maptilerApiKey` param while rendering with Maplibre, but we fail [here](https://github.com/stamen/maperture/blob/main/src/components/GlMap.svelte#L68) because it looks at the URL (without accounting for the API key) before the style requested from the normalized the URL.

**tl;dr**
- I think [this line](https://github.com/stamen/maperture/blob/main/src/components/GlMap.svelte#L68) should privilege `style` over `url` (I don't think there's a side effect of this?) like this:
  ```
  map.setStyle(style || url);
  ```
- As a followup, we should consider normalizing the URLs using the Mapbox protocol _within_ a stylesheet to allow it to be rendered outside of the Mapbox renderer.


Let me know if this does not sound accurate!

Reply:
Changing that line fixed it, but not fully, since there were still errors occuring - for example in the map constructor. I fixed it differently